src/components/Swipe/ProfileCards.js

```diff
@@ -15,48 +15,38 @@ function ProfileCards({ user }) {
 
   useEffect(() => {
     // where the code runs
-    let users = [];
+
+    var users = [];
     var usersRef = db.ref("Users");
+
     usersRef.once("value").then((snapshot) => {
-      for (let i in snapshot) {
-        users.push(i);
-      }
-      for (let i in snapshot.val()) {
-        users.push(i);
-        db.ref("Users/" + i + "/personalInfo/name")
-          .once("value")
-          .then((snapshot) => {
-            // users.push({ key: i, name: snapshot.val() });
+      snapshot.forEach((child) => {
+        try {
+          users.push({
+            key: child.key,
+            name: child.val().personalInfo.name,
+            work: child.val().personalInfo.work,
+            education: child.val().personalInfo.education,
           });
-      }
+        } catch (Error) {
+          //if attributes in databased are missing / named incorrectly
+        }
+      });
       setPeople(users);
     });
 
     //this will run once when component loads and never again
   }, []);
 
-  const getName = (key) => {
-    db.ref("Users/" + key + "/personalInfo/name")
-      .once("value")
-      .then((snapshot) => {
-        return snapshot.val();
-      });
-  };
-
   const onSwipe = (direction, key) => {
     if (direction == "left") {
       //update dont swipe
     }
 
     if (direction == "right") {
-      /*  db.ref("Users/" + user + "/likes").update(
-        "likes",
-        firebase.firestore.FieldValue.arrayUnion(key)
-      );
-      db.ref("Users/" + key + "/likedBy").update(
-        "likedBy",
-        firebase.firestore.FieldValue.arrayUnion(user)
-      );*/
+      db.ref("Users/" + user + "/likes").push(key);
+
+      db.ref("Users/" + key + "/likedBy").push(user);
```

Comment:
Should this be user.uid, rather than user? B/c we wouldn't want to put the entire user object in their likedby right?

Reply:
changed to uid